### PR TITLE
[Order] Adjustment model improvement

### PR DESCRIPTION
**Pull request by michalmarcinkowski**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets |
| License       | MIT
| Doc PR        |

Improvements on Adjustment model to ensure proper relation handling and totals on Adjustable entities.

PR based on #3842.

### CODE REVIEW

**Comment by michalmarcinkowski** [comment]
@pjedrzejewski waiting for review and merge :wink:

File: src/Sylius/Component/Order/spec/Model/AdjustmentSpec.php [hunk 1]
```diff
@@ -148,6 +190,24 @@ function its_neutrality_is_mutable()
...
+    }
+
+    function it_does_not_recalculate_adjustments_on_adjustable_entity_when_neutral_set_to_current_value(
+        OrderInterface $order
+    ) {
+        $order->addAdjustment($this->getWrappedObject())->shouldBeCalled();
+        $this->setAdjustable($order);
+        $order->recalculateAdjustmentsTotal()->shouldNotBeCalled();
+        $this->setNeutral(false);
+        $this->setNeutral(false);
```
Comment (stloyd) [inline src/Sylius/Component/Order/spec/Model/AdjustmentSpec.php]: Duplicated line.

**Comment by pjedrzejewski** [comment]
@michalmarcinkowski Please remove the duplicated line caught by @stloyd and this is good to go!

**Comment by michalmarcinkowski** [comment]
Yeah, this line was redundant. Ready to merge.

**Comment by pjedrzejewski** [comment]
Thanks Michal! :+1:

**Pull request merged**